### PR TITLE
CASM-4225 CASM-4226 Enhancing log messages for IUF consumption

### PR DESCRIPTION
**Pull request by sravani-sanigepalli**
## Summary and Scope

Changes in this PR address the following use cases for IUF -

As an admin, I want to have more contextual information from the result of external operations - `loftsman-manifest-upload and s3-upload`
As an admin, I want to have better logging information about what failed and what I need to fix

In the process, modified `loftsman-manifest-upload` script to make upload operation in a more modularized way (Removed repetition)

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [CASM-4225](https://jira-pro.its.hpecorp.net:8443/browse/CASM-4225)
* Resolves [CASM-4226](https://jira-pro.its.hpecorp.net:8443/browse/CASM-4226)

## Testing

Will be tested on Gamora once the new images are ready

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)? 
- Were continuous integration tests run? If not, why? Yes
- Was upgrade tested? If not, why? NA
- Was downgrade tested? If not, why? NA
- Were new tests (or test issues/Jiras) created for this change? No

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [x] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: bin/loftsman-manifest-upload [hunk 1]
```diff
@@ -66,74 +66,61 @@ process(){
...
-            return 1
+        result=$(cray artifacts create config-data $s3_manifest_key $manifest 2>&1)
+        if [ $? -ne 0 ]; then
+            echo "ERROR Failed to create s3://$s3_manifest_path"
+            result=$(echo "$result" | sed -e 's/^/DEBUG /')
+            echo -e "DEBUG <cray artifacts create config-data $s3_manifest_key $manifest> failed with\n\n$result"
+            exit 1
         else
-            echo "NOTICE successfully created s3://$s3_manifest_path"
+            echo "NOTICE Successfully created s3://$s3_manifest_path"
```
Comment (JohnNavitskyHPE) [inline bin/loftsman-manifest-upload]: this is probably better at INFO

File: bin/s3-upload [hunk 2]
```diff
@@ -75,25 +75,28 @@ process(){
...
-    if ! cray artifacts create $bucket $key $local_path; then
+    result=$(cray artifacts create $bucket $key $local_path 2>&1)
+    if [ $? -ne 0 ]; then
         echo "ERROR Error uploading to S3"
+        result=$(echo "$result" | sed -e 's/^/DEBUG /')
+        echo -e "DEBUG <cray artifacts create $bucket $key $local_path> failed with\n\n$result"
         return 1
     else
-        echo "NOTICE successfully created s3://$bucket/$key"
+        echo "NOTICE Successfully created s3://$bucket/$key"
```
Comment (JohnNavitskyHPE) [inline bin/s3-upload]: info

File: bin/s3-upload [hunk 3]
```diff
@@ -113,7 +116,7 @@ for (( i=0; i< S3_ENTRIES; i++ )); do
     LOCAL_PATH="$PARENT_PATH"/$(echo "$JSON_CONTEXT" | jq -r '.content.s3['$i'].path')
     S3_BUCKET=$(echo "$JSON_CONTEXT" | jq -r '.content.s3['$i'].bucket')
     S3_KEY=$(echo "$JSON_CONTEXT" | jq -r '.content.s3['$i'].key')
-    echo "NOTICE Processing file at $LOCAL_PATH (s3_bucket=$S3_BUCKET s3_key=$S3_KEY)"
+    echo "INFO Processing file at $LOCAL_PATH (s3_bucket=$S3_BUCKET s3_key=$S3_KEY)"
```
Comment (JohnNavitskyHPE) [inline bin/s3-upload]: it would be better to have this as debug, and have an info message based on the success of the operation

**Review by JohnNavitskyHPE — APPROVED** [review]

File: bin/loftsman-manifest-upload (hunk 1 above)
Comment (sravani-sanigepalli) [reply bin/loftsman-manifest-upload]: Updated

**Review by heemstra — APPROVED** [review]

**Pull request merged**